### PR TITLE
Fix: Adjust <Surface/> in order to avoid use any type

### PR DESCRIPTION
**Pull request by LinconOssa**
Enhancing Code Quality and Security: Refactoring to Eliminate Unrestricted Use of Types

### CODE REVIEW

File: src/pages/people/outlets/surfaces/form/RenderContentFormSurface/index.tsx [hunk 1]
```diff
@@ -6,12 +6,15 @@ import { inube } from "@inube/design-system";
...
+import {
+  surfaceFormsConfig,
+  surfaceMessagesConfig,
+} from "../../config/surface.config";
 
 interface RenderSurfaceContentFormProps {
   formType: Appearance;
   handleSubmit: (props: IHandleSubmitProps) => void;
-  surfaceConfig: any;
+  surfaceConfig: typeof surfaceFormsConfig;
```
Comment (carin91) [inline src/pages/people/outlets/surfaces/form/RenderContentFormSurface/index.tsx:17]: i strongly doubt this codeline, when you call the <RenderSurfaceContentForm/> in <Surface/> you are using surfaceFormsConfig, this means that any change that you do to the surfaceFormsConfig will be accepted.

i would prefer the any type. Why is it necessary? Can you find literature from typestript that supports this thought?

File: src/pages/people/outlets/surfaces/interface.tsx [hunk 2]
```diff
@@ -58,10 +59,10 @@ export function SurfacesUI(props: SurfaceUIProps) {
                 type={typeTabs ? "select" : "tabs"}
               />
               {colorTabs.map(
-                (formType: any) =>
-                  selectedTab === formType! && (
+                (formType) =>
```
Comment (carin91) [inline src/pages/people/outlets/surfaces/interface.tsx:62]: Why did you delete the type? Why didn't you use the string type?

File: src/pages/people/outlets/surfaces/form/RenderContentFormSurface/interface.tsx [hunk 3]
```diff
@@ -118,7 +118,7 @@ function RenderSurfaceContentFormUI(props: RenderSurfaceContentFormUIProps) {
               autoColumns="unset"
               autoRows="unset"
             >
-              {surfaceCards.map(([key, config]: any) => (
```
Comment (carin91) [inline src/pages/people/outlets/surfaces/form/RenderContentFormSurface/interface.tsx:121]: Doesn't there need some kind of type assignment?

**Review by carin91 — CHANGES_REQUESTED** [review]

File: src/pages/people/outlets/surfaces/form/RenderContentFormSurface/index.tsx (hunk 1 above)
Comment (LinconOssa) [reply src/pages/people/outlets/surfaces/form/RenderContentFormSurface/index.tsx:17]: 1.surfaceConfig: any;
In this line, you are declaring a surfaceConfig variable of type any. The type any in TypeScript means that the variable can have any type of value. This gives you flexibility, but can also make your code less type-safe, since TypeScript will not apply type checks for this variable. Using any is generally considered less preferable in TypeScript, as you lose the benefits of type checking.

2.surfaceConfig: typeof surfaceFormsConfig;
In this line, you are declaring a surfaceConfig variable and assigning it the type of surfaceFormsConfig using typeof. This means that surfaceConfig should have the same type as surfaceFormsConfig. This provides more type safety, as TypeScript will now perform type checks to ensure that surfaceConfig is used according to the structure of surfaceFormsConfig.

https://medium.com/@juli1pb/why-you-should-not-use-the-type-any-in-typescript-500b27f82a7c
https://www.typescriptlang.org/

File: src/pages/people/outlets/surfaces/interface.tsx (hunk 2 above)
Comment (LinconOssa) [reply src/pages/people/outlets/surfaces/interface.tsx:62]: I deleted it because it is not required since typescript is inferring it automatically
image

File: src/pages/people/outlets/surfaces/form/RenderContentFormSurface/interface.tsx (hunk 3 above)
Comment (LinconOssa) [reply src/pages/people/outlets/surfaces/form/RenderContentFormSurface/interface.tsx:121]: To improve security and code clarity, it would be preferable to specify the exact types of key and config instead of using any. For example:
image
image

**Review by carin91 — APPROVED** [review]
